Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp

```diff
@@ -56,18 +56,19 @@ bool SystemSettingsManagerProxy::darkMode() const
         return true;
 
     // FIXME: These are just heuristics, we should get the dark mode from libhandy/libadwaita, falling back to the settings portal.
-
-    if (auto* themeNameEnv = g_getenv("GTK_THEME"))
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-        return g_str_has_suffix(themeNameEnv, "-dark") || g_str_has_suffix(themeNameEnv, "-Dark") || g_str_has_suffix(themeNameEnv, ":dark");
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+    // Or maybe just use the settings portal, because we don't want to depend on libhandy, and maybe don't want to depend on libadwaita?
+    if (const char* themeNameEnv = g_getenv("GTK_THEME")) {
+        String themeNameEnvString = String::fromUTF8(themeNameEnv);
```

Comment:
I think you're right, because there is an implicit conversion from String to StringView, and I'm using StringView functions anyway, so this will save a bunch of extra conversions.

Reply:
Updated.